libraries/standard/http/utest/http_utest.c

```diff
@@ -59,8 +80,77 @@
+static _headers_t expectedHeaders = { 0 };
+static int testRangeStart = 0;
+static int testRangeEnd = 0;
 static uint8_t httpBuffer[ HTTP_TEST_BUFFER_LEN ] = { 0 };
 
+/* ============================ Helper Functions ============================== */
+
+/**
+ * @brief Fills the test input buffer and expectation buffers with pre-existing data
+ * before calling the API function under test.
+ */
+static void setupBuffersWithPreexistingHeader( HTTPRequestHeaders_t * testRequestHeaders,
+                                               uint8_t * testBuffer,
+                                               size_t bufferSize,
+                                               _headers_t * expectedHeaders,
+                                               const char * preexistingData )
+{
+    size_t dataLen = strlen( preexistingData );
+
+    testRequestHeaders->pBuffer = testBuffer;
+    testRequestHeaders->bufferLen = bufferSize;
+    /* We add 1 bytes as snprintf() writes a null byte at the end. */
+    int numBytes = snprintf( ( char * ) testRequestHeaders->pBuffer,
+                             dataLen + 1,
+                             "%s",
+                             preexistingData );
+    TEST_ASSERT( numBytes == ( int ) dataLen );
+    testRequestHeaders->headersLen = dataLen;
+
+    /* Fill the same data in the expected buffer as HTTPClient_AddRangeHeaders()
+     * is not expected to change it. */
+    TEST_ASSERT( memcpy( expectedHeaders->buffer, testRequestHeaders->pBuffer,
+                         testRequestHeaders->headersLen )
+                 == expectedHeaders->buffer );
+    expectedHeaders->dataLen = testRequestHeaders->headersLen;
+}
+
+/**
+ * @brief Common utility for adding the expected range string for a AddRangeRequest test case
+ * in the expectation buffer.
+ */
+static void addRangeToExpectedHeaders( _headers_t * expectedHeaders,
+                                       const char * expectedRange,
+                                       bool terminatorExists )
+{
+    size_t rangeRequestLen = RANGE_REQUEST_HEADER_FIELD_LEN +
+                             HTTP_HEADER_FIELD_SEPARATOR_LEN +
+                             RANGE_REQUEST_HEADER_VALUE_PREFIX_LEN +
+                             strlen( expectedRange ) +
+                             2 * HTTP_HEADER_LINE_SEPARATOR_LEN;
+    int numBytes =
+        snprintf( ( char * ) expectedHeaders->buffer +
+                  expectedHeaders->dataLen -
+                  ( terminatorExists ? HTTP_HEADER_LINE_SEPARATOR_LEN : 0 ),
+                  /* We add 1 bytes as snprintf() writes a null byte at the end. */
+                  rangeRequestLen + 1,
+                  "%s%s%s%s\r\n\r\n",
+                  RANGE_REQUEST_HEADER_FIELD,
+                  HTTP_HEADER_FIELD_SEPARATOR,
+                  RANGE_REQUEST_HEADER_VALUE_PREFIX,
+                  expectedRange );
+
+    TEST_ASSERT_EQUAL( ( size_t ) numBytes, rangeRequestLen );
```

Comment:
`rangeRequestLen` is expected while `numBytes` is actual so I think this needs to be
```c
TEST_ASSERT_EQUAL( rangeRequestLen, ( size_t ) numBytes );
```

Reply:
Good point. Will fix the order